### PR TITLE
privilege: limit the privileges in memory schemas

### LINKED ISSUE #35205: tidb: client stuck when create table in metrics_schema
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
use metrics_schema;
create table t(id int);
```

```
mysql> mysql> kill 7139;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql>
mysql>
mysql> show processlist;
+------+------+-----------------+----------------+---------+------+------------+------------------------+
| Id   | User | Host            | db             | Command | Time | State      | Info                   |
+------+------+-----------------+----------------+---------+------+------------+------------------------+
| 7141 | root | 127.0.0.1:53157 | NULL           | Query   |    0 | autocommit | show processlist       |
| 7139 | root | 127.0.0.1:53044 | metrics_schema | Query   |  489 | autocommit | create table t(id int) |
+------+------+-----------------+----------------+---------+------+------------+------------------------+
2 rows in set (0.00 sec)
```

### 2. What did you expect to see? (Required)
success if tidb permit it, or fail
### 3. What did you see instead (Required)
stuck, and cannot be killed
### 4. What is your TiDB version? (Required)
`tiup playground v5.4.0 --host 0.0.0.0 --without-monitor --tiflash 0`

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #35205

Problem Summary:

In the previous implementation, `metrics_schema.t` isn't one of the metrics tables, so the privilege checker is skipped. However, because the memory databases are not persist in storage, DDL hangs because of "database not exists" error.

### What is changed and how it works?

This PR disallows users run `CREATE/ALTER/DROP/INSERT/UPDATE/DELETE` statements on **any** tables in memory databases.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

After this PR, the following operations on memory tables are not supported anymore:
- `information_schema`: `REFERENCES`, `EXECUTE`, `SHOW VIEW`, `LOCK TABLES`.
- `performance_schema`: `CREATE`, `LOCK TABLES`.
- `metrics_schema`: `CREATE`, `CREATE VIEW`. `REFERENCES`, `EXECUTE`, `SHOW VIEW`, `LOCK TABLES`.

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix an issue that client stuck when create table in metrics_schema.
```

### CODE REVIEW

File: privilege/privileges/privileges.go [hunk 1]
```diff
@@ -136,31 +135,15 @@ func (p *UserPrivileges) RequestVerification(activeRoles []*auth.RoleIdentity, d
 		}
 	}
 
-	switch dbLowerName {
-	case util.InformationSchemaName.L:
+	if util.IsMemDB(dbLowerName) {
```
Comment (djshow832) [inline privilege/privileges/privileges.go:138]: This PR also disallows creating views on `performance_schema` and `metrics_schema`.

Comment (tangenta) [reply privilege/privileges/privileges.go:138]: MySQL cannot create views in performance_schema.

```sql
mysql> create view performance_schema.v as select * from test.t;
ERROR 1044 (42000): Access denied for user 'root'@'%' to database 'performance_schema'
```

**Review by djshow832 — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**